### DISCUSSION TITLE
after successfully? installing firmware, bluetooth button doesn't work?

### DISCUSSION THREAD

**Discussion by jedierikb** [pybricks/support#1187] [Q&A] [answered]
_Thank you for pybricks! I've been coaching FLL a few years and this year the team is ready to jump into python._

With lego firmware installed, I can connect with my bluetooth dongle.

After installing pybricks firmware (to a yellow spike prime), there is no 'success' or well... anything. I guess it worked?

But, once pybricks firmware is installed, the bluetooth button no longer flashes when pressed. 

Also, windows bluetooth can no longer find the hub.

Suggestions?  Much appreciated.

**Comment by dlech** [comment]
With the Pybricks firmware, Bluetooth is always active when you turn the hub on and you should be able to connect to it using Pybricks Code (by clicking the Bluetooth button in Pybricks Code - do not attempt to use the operating system Bluetooth dialogs to connect). If it doesn't work see #270 for tips on how to collect logs for further investigation.

**Reply by jedierikb** [reply]
Thank you for the quick response.
Alas, after pressing the bluetooth button in the pybricks interface and multiple presses of 'Re-scan' and multiple reboots of the brick, all I get is 'No compatible devices found'.

Per the link you sent me, I tried rebooting windows, but I get the same results - 'No compatible devices found'.

I enabled logging per the link instructions, but nothing looks very suspicious there:
https://photos.app.goo.gl/iuYQfzyHeEx1WjPw7

Some notes: The brick, after the firmware installation, shows a 3x3 square on its display panel. It does not make any sounds.

Nothing can detect it - neither my nearby macbook, nor my iphone... I am not sure it is making itself available for bluetooth pairing?

Thank you for your assistance.

**Reply by dlech** [reply]
Thanks for the logs. The abort errors look suspicious to me. I'm guessing that something changed either in Windows or the browser that broke scanning.

**Reply by laurensvalk** [reply, marked as answer]
The brick seems to be showing the expected behavior. (As a separate improvement, we are considering to just turn on the Bluetooth light to make this more intuitive).

Based on the picture you shared, it does appear that the browser gets a error when it tries to scan for devices. 

The MacBook should be supported when using Google Chrome and connecting via the Bluetooth button in the Pybricks Code app. iPhone is not supported since the iOS browser doesn’t support Bluetooth.

In all cases, connecting can only be done via the Pybricks Code website (not via the phone/computer’s settings menu.)

**Reply by jedierikb** [reply]
the chrome source is not very helpful without way more knowledge about chrome:
https://chromium.googlesource.com/chromium/src/+/5263364e3a1dd453630cfe1df568b5027c38b72d/device/bluetooth/bluetooth_adapter_winrt.cc#938

The errors I was getting were on a windows 10 machine which the FLL team uses.

I just now tried connecting via the browser with my mac. Success!

I guess it is time to get a new windows machine or new bluetooth dongle or get the team a mac.

THANK YOU.

**Reply by dlech** [reply]
If you want to troubleshoot the Windows machine further, we have a Pybricks fan on the MS Bluetooth team. See https://github.com/pybricks/support/issues/918#issuecomment-1527996980 for instructions on how to submit a report.
